### DISCUSSION TITLE
Waiting for a process spawned by `dcli exec`

### DISCUSSION THREAD

**Discussion by sbaldrich** [Dashlane/dashlane-cli#262] [Q&A] [answered]
Sorry if this is a dumb question but I've struggled with it for a couple of hours to no avail. 

Given this `script.sh` file: 

```sh
sleep 100
```

Why does `dcli exec -- ./script.sh` not block until `sleep` finishes? I'm not actually intending to use `sleep` in my script but this reflects the problem I'm facing (I'm actually calling an interactive process)

I initially thought that it had to do with [spawn](https://github.com/Dashlane/dashlane-cli/blob/1df0c618246a990cf44b65df15a591823be5e0dd/src/command-handlers/exec.ts#L21-L25) not [blocking](https://nodejs.org/api/child_process.html#child_processspawnsynccommand-args-options) but I've failed to reproduce it using a minimal example (execution waits until sleep returns):

```js
// app.js , run it using 'node app.js'
const spawn = require('child_process').spawn

const child = spawn('sleep',['10'], {
    stdio: 'inherit',
    shell: true
})

child.on('exit', c => console.log(c))
```

**Comment by Mikescops** [comment]
It's actually a very interesting question and I must admit I haven't tested that many cases yet with the exec command.
As I'm looking to include test scenarios with the CLI I will take more time to look at what you shared and give you a proper answer.
Thanks for sharing.

**Reply by Mikescops** [reply]
I think I see the problem tho'

The CLI command is clearly not waiting for the child event to execute and closes immediately which is a regression that might have been introduced by recent changes. 

@sbaldrich this is definitely worth opening an issue if you're okay to do so 👍

**Reply by Mikescops** [reply, marked as answer]
I need to test more but updating with this seems to work:

```
    const child = spawnSync(command, {
        stdio: 'inherit',
        shell: true,
        env: environmentVariables,
    });

    if (child.error) {
        logger.error(`Failed to execute command: ${child.error.message}`);
    }
```

That way the parent program is forced to wait for the child to exit.

**Comment by sbaldrich** [comment]
Thanks a lot for the [super fast response](https://github.com/Dashlane/dashlane-cli/issues/263)!

**Reply by Mikescops** [reply]
You're welcome!

**Reply by Mikescops** [reply]
I'll tag a new version in the next days.

**Reply by Mikescops** [reply]
Released in v6.2424.2
